### ISSUE TITLE
Simulation `eval_coords` bug

### ISSUE THREAD

**Issue by j-c-c** [ComputationalCryoEM/ASPIRE-Python#823]
When running the notebook in `gallery/tutorials/cov3d_simulation.py` using a `LegacyVolume` with `C`>2 volumes, an error is raised when performing `Simulation.eval_coords`: 

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[7], line 22
     17 # Assign the cluster centroids to the different images. Since we expect a discrete distribution of volumes
     18 # (and therefore of coordinates), we assign the centroid coordinate to each image that belongs to that cluster.
     19 # Evaluate the coordinates estimated
     21 clustered_coords_est = centers[vol_idx]
---> 22 coords_perf = sim.eval_coords(mean_est, eigs_est_trunc, clustered_coords_est)

File ~/Work/ASPIRE-python.simulation_default_volume_811/src/aspire/source/simulation.py:438, in Simulation.eval_coords(self, mean_vol, eig_vols, coords_est)
    435 # 0-indexed states vector
    436 states = self.states - 1
--> 438 coords_true = coords_true[states]
    439 res_norms = res_norms[states]
    440 res_inners = res_inners[:, states]

IndexError: index 2 is out of bounds for axis 0 with size 2
```
 In this case, the `states` indices are being applied to the wrong axis of `coords_true`. This can be fixed by setting `coords_true = coords_true[:, states]`, but this does not work for `C=2` (likely due to the `squeeze()` used in `Simulation.vol_coords').

**Comment by j-c-c** [comment]
One additional observation that was made while tinkering with this notebook is that `scipy.cluster.vq.kmeans2` returns the correct clusters, but sometimes with incorrect indices. For example, `vol_idx` returned by `kmeans2` might be
```
vol_idx = [0, 0, 1, 2, 2]
```
while `sim.states - 1` is
```
sim.states - 1 = [1, 1, 2, 0, 0]
```

**Comment by garrettwrong** [comment]
If you believe that the label of the clustering is not deterministic then we just need a small algo that can effectively check that _a_ clustering matches another clustering.  You can do this by adding corresponding values to a dict on first occurrence. When you encounter again, check the mapping holds. If you make to the end with no conflicts and both sets are same length, I think you're good.

**Comment by j-c-c** [comment]
This issue has been partially resolved by using `sklearn.metrics.adjusted_rand_score` to find the accuracy of the kmeans clustering (#873). 

The tutorial will still only run for `C=2`.

**Comment by j-c-c** [comment]
Closing. Resolved by #1058.
